modules/react/common/lib/responsive/useResponsiveContainerStyles.ts

```diff
@@ -0,0 +1,122 @@
+import { useTheme, isWithinBreakpoint, breakpointKeys } from "@workday/canvas-kit-react/common";
+import type {
+  FlexStyleProps,
+  GridProps
+} from "@workday/canvas-kit-react/layout";
+
+export type BreakpointKeys = typeof breakpointKeys[number];
+
+// TODO: export just all the style props
+type AllStyleProps = FlexStyleProps & GridProps;
+type ResponsiveCSSObject<T> = {
+  [P in keyof T]: Partial<Record<BreakpointKeys, AllStyleProps>> &
+    AllStyleProps;
+};
+type CSSObject<T> = {
+  [P in keyof T]: AllStyleProps;
+};
+
+const getSize = (width: number, breakpoints: any) => {
+  const ranges: {[key: string ]: [number, number?]} = {
+    'zero': [0, breakpoints.s],
+    's': [breakpoints.s, breakpoints.m],
+    'm': [breakpoints.m, breakpoints.l],
+    'l': [breakpoints.l, breakpoints.xl],
+    'xl': [breakpoints.xl]
+  };
+  const size = breakpointKeys.find((size: BreakpointKeys) => isWithinBreakpoint(width, ...ranges[size]))
+
+  return size as BreakpointKeys;
```

Comment:
`find` can return `undefined` if nothing comes back. Line 29 wouldn't catch that because of casting it as `BreakpointKeys`.